components/Chips/src/MDCChipField.m

```diff
@@ -0,0 +1,573 @@
+                    frame.origin.y,
+                    frame.size.width - self.textInsets.left,
+                    frame.size.height);
+}
+
+@end
+
+@interface MDCChipField ()
+    <MDCChipTextFieldDelegate, MDCTextInputPositioningDelegate, UITextFieldDelegate>
+@end
+
+@implementation MDCChipField {
+  NSMutableArray<MDCChipView *> *_chips;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    _chips = [NSMutableArray array];
+    _delimiter = MDCChipFieldDelimiterDefault;
+    _minTextFieldWidth = MDCChipFieldDefaultMinTextFieldWidth;
+    _contentEdgeInsets = MDCChipFieldDefaultContentEdgeInsets;
+    _showPlaceholderWithChips = YES;
+    _chipHeight = 32.0f;
+
+    MDCChipTextField *chipTextField = [[MDCChipTextField alloc] initWithFrame:self.bounds];
+    chipTextField.underline.hidden = YES;
+    chipTextField.delegate = self;
+    chipTextField.deletionDelegate = self;
+    chipTextField.positioningDelegate = self;
+    chipTextField.accessibilityTraits = UIAccessibilityTraitNone;
+    chipTextField.autocorrectionType = UITextAutocorrectionTypeNo;
+    chipTextField.autocapitalizationType = UITextAutocapitalizationTypeNone;
+    chipTextField.keyboardType = MDCChipFieldDefaultKeyboardType;
+    [chipTextField addTarget:self
+                      action:@selector(textFieldDidChange)
+            forControlEvents:UIControlEventEditingChanged];
+    [self addSubview:chipTextField];
+    _textField = chipTextField;
+  }
+  return self;
+}
+
+- (void)setBackgroundColor:(UIColor *)backgroundColor {
+  [super setBackgroundColor:backgroundColor];
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  // Calculate the frames for all the chips and set them.
+  NSArray *chipFrames = [self chipFramesForSize:self.bounds.size];
+  for (NSUInteger index = 0; index < _chips.count; index++) {
+    MDCChipView *chip = _chips[index];
+    chip.frame = [chipFrames[index] CGRectValue];
+  }
+
+  // Get the last chip frame and calculate the text field frame from that.
+  CGRect lastChipFrame = [chipFrames.lastObject CGRectValue];
+  self.textField.frame = [self frameForTextFieldForLastChipFrame:lastChipFrame
+                                                   chipFieldSize:self.bounds.size];
+
+  [self updateTextFieldPlaceholderText];
+}
+
+- (void)updateTextFieldPlaceholderText {
+  // Place holder label should be hidden if showPlaceholderWithChips is NO and there are chips.
+  // MDCTextField sets the placeholderLabel opacity to 0 if the text field has no text.
+  self.textField.placeholderLabel.hidden =
+      (!self.showPlaceholderWithChips && self.chips.count > 0) || ![self isTextFieldEmpty];
+}
+
++ (UIFont *)textFieldFont {
+  return [UIFont systemFontOfSize:[UIFont systemFontSize]];
+}
+
+- (CGSize)intrinsicContentSize {
+  CGFloat minWidth =
+      MAX(self.minTextFieldWidth + self.contentEdgeInsets.left + self.contentEdgeInsets.right,
+          self.bounds.size.width);
```

Comment:
Nit: Please standardize bounds before accessing the size directly, or use `CGRectGetWidth()`